runtime/src/main/java/org/corfudb/runtime/clients/LogUnitHandler.java

```diff
@@ -0,0 +1,207 @@
+            throws Exception {
+        throw new TrimmedException();
+    }
+
+    /**
+     * Handle an ERROR_OVERWRITE message.
+     *
+     * @param msg Incoming Message
+     * @param ctx Context
+     * @param r   Router
+     * @throws OverwriteException Throws OverwriteException if address has already been written to.
+     */
+    @ClientHandler(type = CorfuMsgType.ERROR_OVERWRITE)
+    private static Object handleOverwrite(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r)
+            throws Exception {
+        throw new OverwriteException();
+    }
+
+    /**
+     * Handle an ERROR_DATA_OUTRANKED message.
+     *
+     * @param msg Incoming Message
+     * @param ctx Context
+     * @param r   Router
+     * @throws OverwriteException Throws OverwriteException if write has been outranked.
+     */
+    @ClientHandler(type = CorfuMsgType.ERROR_DATA_OUTRANKED)
+    private static Object handleDataOutranked(CorfuMsg msg,
+                                              ChannelHandlerContext ctx, IClientRouter r)
+            throws Exception {
+        throw new DataOutrankedException();
+    }
+
+
+    /**
+     * Handle an ERROR_VALUE_ADOPTED message.
+     *
+     * @param msg Incoming Message
+     * @param ctx Context
+     * @param r   Router
+     */
+    @ClientHandler(type = CorfuMsgType.ERROR_VALUE_ADOPTED)
+    private static Object handleValueAdoptedResponse(CorfuPayloadMsg<ReadResponse> msg,
+                                                     ChannelHandlerContext ctx, IClientRouter r) {
+        throw new ValueAdoptedException(msg.getPayload());
+    }
+
+    /**
+     * Handle an ERROR_OOS message.
+     *
+     * @param msg Incoming Message
+     * @param ctx Context
+     * @param r   Router
+     * @throws OutOfSpaceException Throws OutOfSpaceException if log unit out of space.
+     */
+    @ClientHandler(type = CorfuMsgType.ERROR_OOS)
+    private static Object handleOos(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r)
+            throws Exception {
+        throw new OutOfSpaceException();
+    }
+
+    /**
+     * Handle an ERROR_RANK message.
+     *
+     * @param msg Incoming Message
+     * @param ctx Context
+     * @param r   Router
+     * @throws Exception Throws Exception if write has been outranked.
+     */
+    @ClientHandler(type = CorfuMsgType.ERROR_RANK)
+    private static Object handleOutranked(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r)
+            throws Exception {
+        throw new Exception("rank");
+    }
+
+    /**
+     * Handle an ERROR_NOENTRY message.
+     *
+     * @param msg Incoming Message
+     * @param ctx Context
+     * @param r   Router
+     * @throws Exception Throws excepton if write is performed to a non-existent entry.
+     */
+    @ClientHandler(type = CorfuMsgType.ERROR_NOENTRY)
+    private static Object handleNoEntry(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r)
+            throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)